### ISSUE TITLE
Implement our own IP address detection 

### ISSUE THREAD

**Issue by SebastianZimmeck** [privacy-tech-lab/privacy-pioneer#362]
At the moment, we are using [ipstack/apilayer](https://ipstack.com/) to get users' IP addresses (IPv4 or IPv6 both work). However, arguably a privacy extension should not send out additional data to a third party. The problem is that it is generally not possible to detect an IP address client-side. So, we need some server. We can decided this at a later point.

A few points from our discussion last Friday:
- Per ipstack, they are only storing the IP address for 30 seconds or so and then delete it.
- If we would be using our own solution, we would still involve a third party (cloud or hosting) provider unless we run our own server. So, switching from one service to the other may only be a gradual difference.
- Sending out the IP address one more time in addition to the third parties that users are disclosing them to may not make much of a difference.
- We currently have implemented a notice and button the user has to push. So, the IP address functionality is not done automatically but is extra functionality with user consent.

**Comment by SebastianZimmeck** [comment]
We decided in our call today that @danielgoldelman will implement everything in our extension that could process the additional data that we get back from Apilayer/Ipstack (e.g., ZIP code or city/region). Once that is done, we decide whether we stick with Apilayer/Ipstack, implement our own version of Apilayer/Ipstack, or do not use such functionality at all.

**Comment by SebastianZimmeck** [comment]
I am closing this issue as it seems to be superseded by #364. Feel free to reopen, @danielgoldelman if I got that wrong.
